cmd/agent/app/snmpwalk.go

```diff
@@ -0,0 +1,316 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gosnmp/gosnmp"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultVersion = ""
+	defaultOID     = ""
+	defaultPort    = 161
+
+	// snmp v1 & v2c
+	defaultCommunityString = "public"
+
+	// snmp v3
+	defaultUserName      = ""
+	defaultAuthProtocol  = ""
+	defaultAuthKey       = ""
+	defaultPrivProtocol  = ""
+	defaultPrivKey       = ""
+	defaultContext       = "public"
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** I think we don't need a default for context name since that's not something users have.

snmpsim uses context name, but it's a very specific to snmpsim tool and doesn't apply to other situations (e.g. customers setup)‏